joomla/joomla-cms
# [3.10] Fix javascript error for pre-update checker

Pull Request for Issue #35406 .

### Summary of Changes
While updating a site from Joomla 3.10 to Joomla 4, if the site contains a system plugin with single quote or double  character in it's name, the pre-update checker will throws some javascript error and update could not be processed

This PR solves it by using `addScriptOptions` instead of `addScriptDeclaration` to pass `nonCoreCriticalPlugins` data to javascript. 

### Testing Instructions
1. Install 3.10.1
2. Install and enable the attached system plugin. (I set the name to **System Fake's** , contains single quote character)
3. Set **Update Channel** to **Joomla Next**
4. Access to Components -> Joomla! Update to try to update from Joomla 3.10 to Joomla 4.
5. Before patch, check browser console, you will see some javascript error
6. After patch, no error, pre-update checker works as expected

[plug_system_fake.zip](https://github.com/joomla/joomla-cms/files/7110155/plug_system_fake.zip)
